src/mlio/image_reader.cxx

```diff
@@ -0,0 +1,413 @@
+        new_width = resize_value;
+    }
+    else {
+        new_height = resize_value;
+        new_width = (resize_value * src.cols) / src.rows;
+    }
+    try {
+        cv::resize(src, dst, cv::Size(new_width, new_height), 0, 0);
+    }
+    catch (cv::Exception const &e) {
+        if (bbh_ != bad_batch_handling::skip) {
+            auto msg = fmt::format(
+                "Image resize operation failed for the image: {0:n} in "
+                "data_store: {1:s}, with the following exception: {2:s}",
+                ins.index(),
+                ins.get_data_store().id(),
+                e.what());
+            if (bbh_ == bad_batch_handling::error) {
+                throw std::runtime_error{msg};
+            }
+            logger::warn(msg);
+        }
+        return false;
+    }
+    return true;
+}
+
+bool
+image_reader::crop(cv::Mat &src, cv::Mat &dst, instance const &ins) const
+{
+    if (src.rows < img_dims_[1] || src.cols < img_dims_[2]) {
+        if (bbh_ != bad_batch_handling::skip) {
+            auto msg = fmt::format(
+                "Input image dimensions [rows: {0:n}, cols: {1:n}] are "
+                "smaller than the output image image_dimensions [rows: {2:n}, "
+                "cols: {3:n}], for the image: {4:n} in data_store: {5:s}",
+                src.rows,
+                src.cols,
+                img_dims_[1],
+                img_dims_[2],
+                ins.index(),
+                ins.get_data_store().id());
+            if (bbh_ == bad_batch_handling::error) {
+                throw std::invalid_argument{msg};
+            }
+            logger::warn(msg);
+        }
+        return false;
+    }
+
+    // center crop
+    int y = (src.rows - img_dims_[1]) / 2;
+    int x = (src.cols - img_dims_[2]) / 2;
+
+    try {
+        cv::Rect roi(x, y, img_dims_[2], img_dims_[1]);
+        src(roi).copyTo(dst);
+    }
+    catch (cv::Exception const &e) {
+        if (bbh_ != bad_batch_handling::skip) {
+            auto msg = fmt::format(
+                "Image crop operation failed for the image: {0:n} in "
+                "data_store: {1:s}, with the following exception: {2:s}",
+                ins.index(),
+                ins.get_data_store().id(),
+                e.what());
+            if (bbh_ == bad_batch_handling::error) {
+                throw std::runtime_error{msg};
+            }
+            logger::warn(msg);
+        }
+        return false;
+    }
+    return true;
+}
+
+}  // namespace v1
+}  // namespace mlio
+
+#else
+
+// clang-format off
+
+#include "mlio/not_supported_error.h"
+
+namespace cv {
+class Mat {};
+}  // namespace cv
+
+namespace mlio {
+inline namespace v1 {
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmissing-noreturn"
+
+image_reader::image_reader(data_reader_params rdr_prm,
+                           image_reader_params image_prm)
+    : parallel_data_reader{std::move(rdr_prm)}, params_{std::move(image_prm)}
```

Comment:
Minor. You can put all member variable initializations to the list.
```
: parallel_data_reader{std::move(rdr_prm)}, params_{std::move(image_prm)}, bbh_{}, img_dims_{}
{
...
}
```

Reply:
resolved as discussed offline.